tests/test_enterprise/api/test_views.py

```diff
@@ -787,6 +795,94 @@ def test_enterprise_customer_catalogs_detail_pagination(self, mock_catalog_api_c
         assert response == expected_result
 
     @ddt.data(
+        (False, {}, {}, None),
+        (False, {}, {'course_run_ids': ['fake1', 'fake2']}, None),
+        (
+            True,
+            {
+                'key': [
+                    fake_catalog_api.FAKE_COURSE_RUN['key'],
+                    fake_catalog_api.FAKE_COURSE_RUN2['key']
+                ]
+            },
+            {
+                'course_run_ids': [
+                    fake_catalog_api.FAKE_COURSE_RUN['key'],
+                    fake_catalog_api.FAKE_COURSE_RUN2['key']
+                ]
+            },
+            None
+        ),
+        (
+            True,
+            {
+                'uuid': [
+                    fake_catalog_api.FAKE_PROGRAM_RESPONSE1['uuid'],
+                    fake_catalog_api.FAKE_PROGRAM_RESPONSE2['uuid']
+                ]
+            },
+            {
+                'program_uuids': [
+                    fake_catalog_api.FAKE_PROGRAM_RESPONSE1['uuid'],
+                    fake_catalog_api.FAKE_PROGRAM_RESPONSE2['uuid']
+                ]
+            },
+            None
+        ),
```

Comment:
@douglashall I think I'm missing something: Why is `search_results` `None` in the two data sets above? The values passed as `course_run_ids` and `program_uuids` match the values listed in `content_filter`, so shouldn't they be part of the `response` from the endpoint (L874 below)?

Reply:
When the `content_filter` specifies unique keys for content items, we only need to check the content_filter to determine if the catalog contains the given content items. We will not make a search request to the discovery service, thus `search_results` is `None` here.

Reply:
Looks good to me after reading the test code itself carefully; I think DDT sometimes backfires in cases like this though, as I've found that it confuses more than it helps, or plainly makes it harder to see what's being tested. Maybe better to split things up into separate classes and tests according to how they share code rather than have huge DDT data sets. I think the tell-tale sign is when you have an if/else statement for testing 2 different scenarios -- those can be 2 tests for sure.

Reply:
@UmanShahzad I agree with your strategy for breaking things into separate tests and the tell-tale if/else sign. I'll break this up. Hopefully we can continue moving in that direction going forward.